client/landing/stepper/declarative-flow/internals/steps-repository/launch-big-sky/index.tsx

```diff
@@ -104,7 +108,9 @@ const LaunchBigSky: Step = function () {
 				setProgress( 75 );
 
 				window.location.replace(
-					`${ siteURL }/wp-admin/site-editor.php?canvas=edit&referrer=design-choices`
+					`${ siteURL }/wp-admin/site-editor.php?canvas=edit&referrer=${
+						referrer || 'design-choices'
```

Comment:
This should be escaped as well, because it's user input.

Reply:
I simplified it much more - we passed the flow name in the launch big sky step.
https://github.com/Automattic/wp-calypso/pull/100948/commits/678834230db0a7215557183c0c3bae970f25915f